### PR TITLE
Login

### PR DESCRIPTION
**Pull request by tayor99**
I fixed previous pr issues and integrated a login and register feature

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -1,38 +1,24 @@
...
+:root {
+  --twitter-blue: #1a8cd8;
+  --little-dark: #585757;
+  --black: #000;
 }
-
-.App-logo {
-  height: 40vmin;
-  pointer-events: none;
+ody {
```
Comment (Asylcreek) [inline src/App.css:6]: `ody`?

File: src/components/SIgnUpForm.js [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+  }, [createUser.password, createUser.password_confirmation]);
+  useEffect(() => {
+    setErrorMessage('');
+  }, [createUser]);
+
+  const handleChange = (e) => {
+    setCreateUser({
+      ...createUser,
+      [e.target.name]: e.target.value,
+    });
```
Comment (Asylcreek) [inline src/components/SIgnUpForm.js:55]: Use the updater function instead

File: src/components/SIgnUpForm.js [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+import axios from 'axios';
+import { BASE_URL } from '../App';
+import { AiOutlineCheckCircle } from 'react-icons/ai';
+import { FaTimesCircle } from 'react-icons/fa';
+
+const PASSWORD_CHECK =
+  /^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#$%]).{8,24}$/;
+const REGISTER_URL = `${BASE_URL}/register`;
+const SIgnUpForm = ({ handleCloseModal }) => {
+  const userRef = useRef();
```
Comment (Asylcreek) [inline src/components/SIgnUpForm.js:12]: Where did you use `userRef`?

File: src/components/SIgnUpForm.js [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+  const [createUser, setCreateUser] = useState({
+    first_name: '',
+    last_name: '',
+    email: '',
+    username: '',
+    password: '',
+    password_confirmation: '',
+  });
+
+  const fieldsCheck =
```
Comment (Asylcreek) [inline src/components/SIgnUpForm.js:23]: You can do better than `fieldsCheck` for this variable name

File: src/components/SignInForm.js [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+  const { setUser } = useContext(UserContext);
+  const [loginDetails, setLoginDetails] = useState({
+    username: '',
+    password: '',
+  });
+  const [errorMessage, setErrorMessage] = useState('');
+
+  useEffect(() => {
+    setErrorMessage('');
+  }, [loginDetails.username, loginDetails.password]);
```
Comment (Asylcreek) [inline src/components/SignInForm.js:19]: Just `loginDetails` would have sufficed

File: src/components/SignInForm.js [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+  const handleChange = (e) => {
+    setLoginDetails({
+      ...loginDetails,
+      [e.target.name]: e.target.value,
+    });
```
Comment (Asylcreek) [inline src/components/SignInForm.js:25]: Use the updater function instead

File: src/styles/sidebar.css [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  width: 230px;
+  height: 50px;
+  background-color: var(--twitter-blue);
+  border-radius: 30px;
+  color: #fff;
+  font-size: 1.2rem;
+  font-family: 'Roboto', sans-serif;
```
Comment (Asylcreek) [inline src/styles/sidebar.css:27]: This should either be in a CSS variable or it should be `font-family: inherit`

File: src/styles/sidebar.css [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+  align-items: center;
+}
+.sidebarUser__profile > img {
+  width: 40px;
+  object-fit: cover;
+  border-radius: 50px;
+  margin-right: 10px;
+}
+.sidebarUser__name > p.username {
+  font-family: 'Roboto', sans-serif;
```
Comment (Asylcreek) [inline src/styles/sidebar.css:50]: This should either be in a CSS variable or it should be `font-family: inherit`

File: src/styles/sidebar.css [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+  font-size: 0.9rem;
+  font-weight: 600;
+}
+.sidebarUser__name > p.name {
+  font-family: 'Roboto', sans-serif;
```
Comment (Asylcreek) [inline src/styles/sidebar.css:55]: This should either be in a CSS variable or it should be `font-family: inherit`

File: src/styles/sidebarnav.css [hunk 10]
```diff
@@ -0,0 +1,28 @@
+.sidebarNav {
+  margin-bottom: 2px;
+}
+.sidebarNav__details {
+  font-family: 'Roboto', sans-serif;
```
Comment (Asylcreek) [inline src/styles/sidebarnav.css:5]: This should either be in a CSS variable or it should be `font-family: inherit`

File: src/App.js [hunk 11]
```diff
@@ -1,24 +1,45 @@
...
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
+    <UserContext.Provider value={{ user, setUser }}>
```
Comment (Asylcreek) [inline src/App.js:26]: Why did you use Context?

File: src/components/SignInForm.js [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+  };
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    try {
+      const { data } = await axios.post(LOGIN_URL, loginDetails);
+      setUser(data.user);
+      localStorage.setItem('loginDetails', JSON.stringify(data));
```
Comment (Asylcreek) [inline src/components/SignInForm.js:33]: If you are only using the user value, why are you storing the whole `data` object in `localStorage`?

**Review by Asylcreek — CHANGES_REQUESTED** [review]

File: src/components/SIgnUpForm.js (hunk 3 above)
Comment (tayor99) [reply src/components/SIgnUpForm.js:12]: i was actually planning on using it but i removed it

File: src/components/SIgnUpForm.js (hunk 4 above)
Comment (tayor99) [reply src/components/SIgnUpForm.js:23]: ok sir

File: src/App.js (hunk 11 above)
Comment (tayor99) [reply src/App.js:26]: I used context to avoid prop drilling

File: src/components/SignInForm.js (hunk 12 above)
Comment (tayor99) [reply src/components/SignInForm.js:33]: i also need to store the access token and refresh token

File: src/App.js (hunk 11 above)
Comment (Asylcreek) [reply src/App.js:26]: Hmmm.

And you want to avoid prop drilling because?

File: src/components/SignInForm.js (hunk 12 above)
Comment (Asylcreek) [reply src/components/SignInForm.js:33]: I see

Comment (tayor99) [reply src/components/SignInForm.js:33]: Sir i have been running into an issue with my api calls i keep getting cors error on other browser except on my main chrome browser and even on my chrome browser i can only call the register and login endpoints i get cors error on other api calls

File: src/App.js (hunk 11 above)
Comment (tayor99) [reply src/App.js:26]: I feel it causes confusion and it is messy

File: src/App.css (hunk 1 above)
Comment (tayor99) [reply src/App.css:6]: This was a mistake sir

File: src/components/SignInForm.js (hunk 5 above)
Comment (tayor99) [reply src/components/SignInForm.js:19]: ok sir

File: src/components/SignInForm.js (hunk 12 above)
Comment (Asylcreek) [reply src/components/SignInForm.js:33]: The CORS error is for your backend to solve

File: src/App.js (hunk 11 above)
Comment (Asylcreek) [reply src/App.js:26]: But you have only two places that need it ... Don't you feel context might be a big leap for your current use case?

Comment (tayor99) [reply src/App.js:26]: I feel i might need to use it in other component the more i work on the project so i thought of setting it up

Comment (Asylcreek) [reply src/App.js:26]: Okay

**Pull request merged**